editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditFeaturesWithFeatureLinkedAnnotationController.java

```diff
@@ -0,0 +1,310 @@
+    EditAttributesDialog editAttributesDialog = new EditAttributesDialog(selectedFeature);
+
+    // show the dialog and wait for the user response
+    editAttributesDialog.showAndWait();
+  }
+
+  /**
+   * Update the attributes of the selected feature.
+   */
+  private void updateAttributes(Feature selectedFeature) {
+
+    // update feature in the feature table
+    ListenableFuture<Void> editResult = selectedFeature.getFeatureTable().updateFeatureAsync(selectedFeature);
+    editResult.addDoneListener(() -> {
+              try {
+                if (editResult.isDone()) {
+                  editResult.get();
+                }
+              } catch (InterruptedException | ExecutionException e) {
+                new Alert(Alert.AlertType.ERROR, "Cannot update attributes: " + e.getCause().getMessage()).show();
+              }
+            }
+    );
+  }
+
+  /**
+   * Move the selected point feature to the given map point by updating the selected
+   * feature's geometry and feature table.
+   */
+  private void movePoint(Point mapPoint) {
+
+    // set the selected features' geometry to a new map point
+    selectedFeature.setGeometry(mapPoint);
+
+    // update the selected feature's feature table
+    updateAttributes(selectedFeature);
+
+    // clear selection of the point
+    clearSelection();
+    selectedFeature = null;
+  }
+
+  /**
+   * Move the last of the vertex point of the currently selected polyline to the given map point by updating the
+   * selected feature's geometry and feature table.
+   */
+  private void movePolylineVertex(Point mapPoint) {
+
+    // get the selected feature's geometry as a polyline
+    Polyline polyline = (Polyline) selectedFeature.getGeometry();
+
+    // create a polyline builder to add and remove parts from the polyline
+    PolylineBuilder polylineBuilder = new PolylineBuilder(polyline);
+
+    // get the nearest vertex to the map point on the polyline
+    ProximityResult nearestVertex = GeometryEngine.nearestVertex(polyline, (Point) GeometryEngine.project(mapPoint, polyline.getSpatialReference()));
+
+    // get the part of the polyline nearest to the map point
+    Part part = polylineBuilder.getParts().get((int) nearestVertex.getPartIndex());
+
+    // remove the nearest point to the map point from the part
+    part.removePoint((int) nearestVertex.getPointIndex());
+
+    // add the map point as the new point on the part
+    part.addPoint((Point) GeometryEngine.project(mapPoint, polyline.getSpatialReference()));
+
+    // add the part to the polyline
+    polylineBuilder.getParts().add(part);
+
+    // set the selected feature's geometry to the new polyline
+    selectedFeature.setGeometry(polylineBuilder.toGeometry());
+
+    // update the selected feature's feature table
+    updateAttributes(selectedFeature);
+
+    // clear selection of the polyline
+    clearSelection();
+    selectedFeatureIsPolyline = false;
+    selectedFeature = null;
```

Comment:
Looks like every time we call `clearSelection()`, we also do `selectedFeature = null`. So it would be possible to put that statement in the method as well.
We could also pop the statement `selectedFeatureIsPolyline = false;` in the method as well, although it's not necessary to set that boolean every time.